### DISCUSSION TITLE
[Unknown Mod][Locations][Creature Generators] Adventure map, right click information of creature generators on the adventure map

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4665] [Ideas]
When at least one of the player's heroes has visited a creature generator on the adventure map, the number of available creatures is displayed when the player right clicks on the generator.

Example with Excavation:
Generator visited by one of the player's heroes, creatures are available:

"(available: 56)"

![Excavation 01](https://user-images.githubusercontent.com/43583994/142608056-94a0dc35-ad60-4ef5-be3e-de1a80972a06.png)

Example with Halfling Hole:
Generator visited by one of the player's heroes, all creatures have been recruited by one of the player's heroes:

"(empty)"

![Generator creature empty 01](https://user-images.githubusercontent.com/43583994/142608134-c702ed80-a0bb-4109-8c89-eb14c1a596f2.png)

However, when a new week starts, the number of creatures is increased and the number of previously available creatures is no longer displayed (the number of available creatures has increased but the player does not know by how much and should not know without revisiting the generator).
This is really not practical.

Also, when an opposing hero has visited a creature generator (already revealed on the map by one of the player's heroes) the information should no longer be available until one of the player's heroes has revisited the creature generator.

In fheroes2:

https://user-images.githubusercontent.com/43583994/142608204-fcb90e90-00bc-487e-92da-3b704a626401.mp4

**Edit 2025/07/13.**

I suggest that the information already known is updated in line with the growth range.

Example with 56 creatures plus one week of growth (ex: 3-6):

"(available: 59-62)"

Example with 0 creatures "(empty)" (one of the player's heroes has already recruited the creatures from the generator before) plus two weeks of growth:

"(available: 6-12)"

Of course, if an opposing hero has visited the creature generator, and even without recruiting creatures, the number of creatures is no longer indicated.
Current week, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes:

Nothing is displayed, as it is now.

With four weeks of growth, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes, only growth is indicated:

"(growth: 12-24)"

Here is the save file (multiplayer):
[Before ruins 01.zip](https://github.com/ihhub/fheroes2/files/7569586/Before.ruins.01.zip)

This is not an option.
This can be implemented directly in fheroes2.

**Comment by zenseii** [comment]
I agree that this should be adjusted, as it seems almost like a bug that you'll stop getting to know how many there are when a new week has started. Why not just add the number of the newly recruitable creatures to the amount of available creatures? It seems like you'll end up with a lot of unnecessary text if you're gonna keep track of how weeks have passed since last you visisted that dwelling.

Essentially this is just to avoid forcing the player to exit the game to look up what weekly growth halflings have and then make the calcuations themself to know how many halflings are currently available, when the game could just simply do that.

Another issue that will need to be dealt with with showing how many weeks is that one of the weeks could be a week of the plague, and so that would depend on the player remembering what week that happened in, even maybe having to write that down while playing, which is a veeery old school way of playing.

**Reply by LeHerosInconnu** [reply]
Hello @zenseii,

> I agree that this should be adjusted, as it seems almost like a bug that you'll stop getting to know how many there are when a new week has started.

Yes, this can be considered a bug, or at least something that is not done in an optimal way for a good gaming experience.

> Why not just add the number of the newly recruitable creatures to the amount of available creatures?

Because the number of creatures added is not constant but is a range, for example, plus 2 to 5 creatures per week.
Also, I have planned an enhancement for the Scouting secondary skill that would give the exact number of creatures: https://github.com/ihhub/fheroes2/issues/1506.

> It seems like you'll end up with a lot of unnecessary text if you're gonna keep track of how weeks have passed since last you visisted that dwelling.
> 
> Essentially this is just to avoid forcing the player to exit the game to look up what weekly growth halflings have and then make the calcuations themself to know how many halflings are currently available, when the game could just simply do that.
>
> Another issue that will need to be dealt with with showing how many weeks is that one of the weeks could be a week of the plague, and so that would depend on the player remembering what week that happened in, even maybe having to write that down while playing, which is a veeery old school way of playing.

You are right, the numbers displayed can also directly include the creatures added by taking into account the range. ;)
- For example, with a range of 2 to 5 creatures and a week of growth, display:

"(available: 58-61)"

Instead of:

"(available: 56)
(+1 week of growth)"

- And:

"(available: 4-10)"

Instead of:

"(was empty)
(+2 weeks of growth)"

- And:

"(at least: 8-20)"

Instead of:

"(unknown)
(+4 weeks of growth)"

- And :

"(empty)" remains "(empty)".

- And :

"(unknown)" remains "(unknown)".

The Plague can only appear at the beginning of the month, it's the "Month of the Plague", there is no "Week of the Plague". :)

**Reply by Branikolog** [reply]
Hello, @LeHerosInconnu & @zenseii !

> It seems like you'll end up with a lot of unnecessary text if you're gonna keep track of how weeks have passed since last you visisted that dwelling.

Totally agree with that.

What if we just simply return to the traditional creature quantity descriptions here?
Once we visit a dwelling we should know exact amount of creatures, that's true. But the next week we can just show approximate description, the same to the neutral creatures on an adventure map. For example: after visiting we should watch 56 at info window. The next week, we can type simply "Horde". Until we visit dwelling once again. In a case the number of creatures exceeds 99, we change the description to "Throng".

After we picked all creatures, next week info window will show "a few", "several" etc.

>Of course, if an opposing hero has visited the creature generator, and even without recruiting creatures, the number of creatures is no longer indicated.
>Current week, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes:
>
>"(unknown)"
>
>With four weeks of growth, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes:
>
>"(unknown)
>(+4 weeks of growth)"

I suggest to return to the basic logic here:
Unknown number - let's show nothing at info window. "Unknown+ x" looks like a school exercises. :)

**Reply by zenseii** [reply]
I think this verbal description could work well too, but I would need to try it to give a final answer since it is a very unusual change for the series

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog and @zenseii,

> Hello, @LeHerosInconnu & @zenseii !
> 
> > It seems like you'll end up with a lot of unnecessary text if you're gonna keep track of how weeks have passed since last you visisted that dwelling.
> 
> Totally agree with that.
> 
> What if we just simply return to the traditional creature quantity descriptions here? Once we visit a dwelling we should know exact amount of creatures, that's true. But the next week we can just show approximate description, the same to the neutral creatures on an adventure map. For example: after visiting we should watch 56 at info window. The next week, we can type simply "Horde". Until we visit dwelling once again. In a case the number of creatures exceeds 99, we change the description to "Throng".
> 
> After we picked all creatures, next week info window will show "a few", "several" etc.

> I think this verbal description could work well too, but I would need to try it to give a final answer since it is a very unusual change for the series

The terms "Few", etc. are too vague.
(And, I think there are already too many "Few", etc. terms in the game: https://github.com/ihhub/fheroes2/issues/1530.) :)

Also the quantity range is known (and the possible number of creatures can be found by deduction), might as well give a clear information about it. :)

> > Of course, if an opposing hero has visited the creature generator, and even without recruiting creatures, the number of creatures is no longer indicated.
> > Current week, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes:
> > "(unknown)"
> > With four weeks of growth, generator visited last by one of the opponent's heroes and not revisited by one of the player's heroes:
> > "(unknown)
> > (+4 weeks of growth)"
> 
> I suggest to return to the basic logic here: Unknown number - let's show nothing at info window. "Unknown+ x" looks like a school exercises. :)

I think it's okay not to display "(unknown)" for the current week, but it's useful to display "(at least: 8-20)" for the following weeks.
As I wrote:

> - And:
> 
> "(at least: 8-20)"
> 
> Instead of:
> 
> "(unknown)
> (+4 weeks of growth)"

**Reply by Branikolog** [reply]
@LeHerosInconnu 

>The terms "Few", etc. are too vague.
>(And, I think there are already too many "Few", etc. terms in the game: #1530.) :)
>
>Also the quantity range is known (and the possible number of creatures can be found by deduction), might as well give a clear information about it. :)

"Few", "Several", etc - are "heroes classics"! You shouldn't be afraid of these terms. :)  All of them are quite descriptive and familiar to players, since the same are used all over the game.

In my personal point, game loses a bit of individuality, when all these terms are converted into "100-500", "5-10", "45+37" etc. Sure, we've discussed before the optional turning on of these features (for newcomers or those, who likes numbers more, probably), but for default view, I think, original quantity description should be left as original and authentic. :)

I agree with you, that numeric values are more descriptive. But at the same time any player, after 1-2 played scenario will easily understand all that "hordes"

>I think it's okay not to display "(unknown)" for the current week, but it's useful to display "(at least: 8-20)" for the following weeks.
>>As I wrote:
>>
>>And:
>>"(at least: 8-20)"
>>
>>Instead of:
>>
>>"(unknown)
>>(+4 weeks of growth)"

At the start of each game we have like more or less predictable values for all dwellings, which could be added as well, regarding this discussion. So in a result all dwellings would have too "numeric" status all over the game if we would type: "50+ 10-40". It's like a school maths lesson. :P
The same to resource piles: we can have like 2-5 resources from each pile, which could be written in a popup window, but should we?
I think game doesn't need excessive specifics for non-specific values.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> @LeHerosInconnu
> 
> > The terms "Few", etc. are too vague.
> > (And, I think there are already too many "Few", etc. terms in the game: #1530.) :)
> > Also the quantity range is known (and the possible number of creatures can be found by deduction), might as well give a clear information about it. :)
> 
> "Few", "Several", etc - are "heroes classics"! You shouldn't be afraid of these terms. :) All of them are quite descriptive and familiar to players, since the same are used all over the game.

I'm not afraid of the terms, it's just that I prefer to see the numbers directly, why complicate things. :)

> In my personal point, game loses a bit of individuality, when all these terms are converted into "100-500", "5-10", "45+37" etc. Sure, we've discussed before the optional turning on of these features (for newcomers or those, who likes numbers more, probably), but for default view, I think, original quantity description should be left as original and authentic. :)

Each player will have a preference here, the one who prefers numbers directly and the one who prefers terms.
I know which group I belong to and I know which group you belong to. :)

> I agree with you, that numeric values are more descriptive. But at the same time any player, after 1-2 played scenario will easily understand all that "hordes"

After all these years of playing, I still sometimes confuse these terms. :)

> > I think it's okay not to display "(unknown)" for the current week, but it's useful to display "(at least: 8-20)" for the following weeks.
> > > As I wrote:
> > > And:
> > > "(at least: 8-20)"
> > > Instead of:
> > > "(unknown)
> > > (+4 weeks of growth)"
> 
> At the start of each game we have like more or less predictable values for all dwellings, which could be added as well, regarding this discussion.

It's a good idea.
I'm 100% for it. 
Not a Few for it, not a Pack for it, not a Horde for it, yes 100% for it. :D

> So in a result all dwellings would have too "numeric" status all over the game if we would type: "50+ 10-40". It's like a school maths lesson. :P

And in the example, how to indicate a range of 2-5 with the terms?

"(few-several)" :D

And for 10-40?

"(Pack-Lots)" :eyeglasses: 

> The same to resource piles: we can have like 2-5 resources from each pile, which could be written in a popup window, but should we? I think game doesn't need excessive specifics for non-specific values.

This is different for resource stacks, etc., the amount of resources in the stack does not change over time and the stacks disappear from the adventure map when picked up.

**Reply by XCOM-HUB** [reply]
It would be cool if the Adventure Map displayed how many creatures are in a Creature Dwelling once you own at least one Thieves’ Guild. It could also show how many resources are in a resource pile—using text descriptions, of course—such as “A few”, “Several”, and so on.

With additional Thieves’ Guilds, the information could become more precise, eventually revealing the exact number of creatures in any Dwelling and the exact amount of resources in any pile. It could even show what resources you will receive from a Windmill each week.

If a Treasure Chest contains a Leprechaun offering an artifact, it could say “A Leprechaun will offer you an artifact.” And with enough Thieves’ Guilds, it could even reveal which artifact the Leprechaun will offer—for example, Fizbin of Misfortune—giving you a heads‑up to avoid that particular chest.

This way, the Thieves’ Guilds could be more useful than they already are.

**Comment by Branikolog** [comment]
>Each player will have a preference here, the one who prefers numbers directly and the one who prefers terms.
I know which group I belong to and I know which group you belong to. 

😄 

>After all these years of playing, I still sometimes confuse these terms. :)

In H3 modders made both appear at the same time. Ex: " Lots (21-49) ". We could probably do the same.

>"(at least: 8-20)"

This sounds a bit not correct. You never say like that in common life. "At least" and "at most" phrases are always stick to a single number.  It should look like " At least 8 ", but not "at least 8-20".  
But I'm still against this idea. :) I find it rather confusing, if I watch "At least 5", but AI didn't pick troops from a dwelling and I get like 60.... 
In this case it would better show "Some", so player could understand, that there're some creatures, rather than tell, that "you can find 5 - ♾️" there...

>And in the example, how to indicate a range of 2-5 with the terms?
>
>"(few-several)" :D
>And for 10-40?

We don't need to indicate a range.
We should indicate an exact number of creatures available in dwelling using all known, common terms.
Ex: player visited a dwelling, where he found 15 creatures of any type. He didn't pick them, and next week, while pointing the dwelling he would see a "Pack", if their number became like 18, or "Lots", if their number became 20, 21... etc.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > After all these years of playing, I still sometimes confuse these terms. :)
> 
> In H3 modders made both appear at the same time. Ex: " Lots (21-49) ". We could probably do the same.

This is certainly used for wandering creatures?
But what about heroes and castles?

> > "(at least: 8-20)"
> 
> This sounds a bit not correct. You never say like that in common life. "At least" and "at most" phrases are always stick to a single number. It should look like " At least 8 ", but not "at least 8-20". But I'm still against this idea. :) I find it rather confusing, if I watch "At least 5", but AI didn't pick troops from a dwelling and I get like 60.... In this case it would better show "Some", so player could understand, that there're some creatures, rather than tell, that "you can find 5 - infinity" there...

I transcribed (to reduce the length of the line) "at least between 8 and 20" to "at least: 8-20".
Meaning there are at least between 8 and 20 creatures available (there may be more).
This is because it is possible to recruit only a part of the creatures available in some of the creature generators on the adventure map.

> > And in the example, how to indicate a range of 2-5 with the terms?
> > "(few-several)" :D
> > And for 10-40?
> 
> We don't need to indicate a range. We should indicate an exact number of creatures available in dwelling using all known, common terms. Ex: player visited a dwelling, where he found 15 creatures of any type. He didn't pick them, and next week, while pointing the dwelling he would see a "Pack", if their number became like 18, or "Lots", if their number became 20, 21... etc.

Let's say that with a growth of 2 to 5 creatures per week, there are effectively 24 creatures available after 8 weeks.

With the range display this gives:

"(available: 16-40)"

Using the terms this gives:

"(Lots)"

This corresponds to 20-49.

Sorry, but I prefer the displayed numbers of the range which give a more accurate evaluation than with the terms.
With the range, I am sure to have at most 40.
With the term, it misleads me into thinking I can have up to 49, more than I can actually have.
It's not precise enough with the terms.

**Reply by Branikolog** [reply]
>But what about heroes and castles?

Don't remember, actually. There're used either words or numbers only...

>With the range display this gives:
>
>"(available: 16-40)"

Forget about the ranges. After 8 weeks there will be an exact actual number of creatures available on visiting in a dwelling. Once you visit the dwelling you'll see, for instance, 41 of Pink Goblins available for hiring, which corresponds term "Lots", which is understandable to everyone.

I'm not a fan of adding random ranges of creature quantity... All these 16-48, 10-53, 71-112 etc. look a bit messy. 
If the majority of players and our team vote for numeric creature description, It would be nice, if creature dwellings will show the same intervals, we use for creatures at enemy armies at least: if there're 6 creatures available - then info window should say "several" or 4-12; for available 15 - "pack" or 12-20 etc.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > With the range display this gives:
> > 
> > "(available: 16-40)"
> > 
> Forget about the ranges. After 8 weeks there will be an exact actual number of creatures available on visiting in a dwelling. Once you visit the dwelling you'll see, for instance, 41 of Pink Goblins available for hiring, which corresponds term "Lots", which is understandable to everyone.

I think you don't get my point. :)

If the player visits the creature generator and therefore knows exactly how many creatures are in the generator, in your example "41 of Pink Goblins", it doesn't make sense to show "Lots" instead of the number since the player knows the number of creatures exactly.
It's like if the player has 3 Titans defending his castle and you display "Few" instead.

> I'm not a fan of adding random ranges of creature quantity... All these 16-48, 10-53, 71-112 etc. look a bit messy.
> If the majority of players and our team vote for numeric creature description, It would be nice, if creature dwellings will show the same intervals, we use for creatures at enemy armies at least: if there're 6 creatures available - then info window should say "several" or 4-12; for available 15 - "pack" or 12-20 etc.

On the adventure map, the creature generators do not have a fixed growth (as is the case for the generators in the castles), except for the Red Dragon (in Dragon City) which always has a growth of "1" and the Genie (in Ancient Lamp) which has no growth.

The growth of the creature generators on the adventure map has a range, e.g. 3-6 for the Rogue (in Wagon Camp).
The possible growth per week is therefore known (3 to 6), but not the exact number (e.g. 4 in the first week, 5 in the second week, etc.) and the number of creatures available can be easily deduced.
If there were 40 rogues available on the first visit, then there will be at least 43 rogues and at most 46 rogues available the following week, and so on for the following weeks.

Nothing is messy in all this, everything is known, the goal is to make it easy for the player by showing him directly with precision what can be inside the creature generator.

**Reply by Branikolog** [reply]
@LeHerosInconnu 

>I think you don't get my point. :)
>
>If the player visits the creature generator and therefore knows exactly how many creatures are in the generator, in your example "41 of Pink Goblins", it doesn't make sense to show "Lots" instead of the number since the player knows the number of creatures exactly.
It's like if the player has 3 Titans defending his castle and you display "Few" instead.

I get it. :)
On a current week it's obvious, that player should see the exact number of creatures available. (till the end of the week or other player visits the dwelling)
I was talking about the NEXT week, after hero visited a dwelling. As we cannot know exact number of creatures available next week, it's enough for player to watch the approximate number with traditional ranges, for example for a "few" (1-4) or "several" (5-12) etc creatures appeared for hiring.

>The possible growth per week is therefore known (3 to 6), but not the exact number (e.g. 4 in the first week, 5 in the second week, etc.) and the number of creatures available can be easily deduced.

And it's still not exact number. And rather than showing X + (Y -Z) I still prefer classical quantity description using human words which is the same throughout all series. "Many", "Not many", "Too many" etc, rather than maths exercices. :smile: 
I think. (I'm not offending, it's my personal point) that game's style will suffer from too excessive "numberisticness". Even game developers  in the original prefered not to show spell damage (which is wrong in my point) to keep a balance within text messages and numerical values of stats, resource values and damage. If we full the game with all precise quantities using numbers it will loose some magic, I'm afraid.

I don't know, why am I so frightened with numbers... Could it be an injury from school lessons? But I loved maths and and always got "A". :rofl: 

I completely understand and accept your proposal. It is more useful and rises comfort of the gameplay, comparing to what I am talking above. But I'm agree to sacrifice a little bit of preciseness of your proposal for the game style and "literary" pleasant-looking words. :)

**Reply by zenseii** [reply]
Just to annoy you guys even more, here's a picture of the official Japanese localization by Imagineer Co.:

![image](https://user-images.githubusercontent.com/12501091/143295362-d0f81863-8306-4585-9ce5-d87aa0a24ecf.png)

Gotta love those 0-4 Ogre Lords. Maybe the hero has them, or maybe he doesn't - who knows? ... Technically, any army has 0-4 Black Dragons too.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog and @zenseii,

> @LeHerosInconnu
> 
> > I think you don't get my point. :)
> > If the player visits the creature generator and therefore knows exactly how many creatures are in the generator, in your example "41 of Pink Goblins", it doesn't make sense to show "Lots" instead of the number since the player knows the number of creatures exactly.
> > It's like if the player has 3 Titans defending his castle and you display "Few" instead.
> 
> I get it. :) On a current week it's obvious, that player should see the exact number of creatures available. (till the end of the week or other player visits the dwelling) I was talking about the NEXT week, after hero visited a dwelling. As we cannot know exact number of creatures available next week, it's enough for player to watch the approximate number with traditional ranges, for example for a "few" (1-4) or "several" (5-12) etc creatures appeared for hiring.
> 
> > The possible growth per week is therefore known (3 to 6), but not the exact number (e.g. 4 in the first week, 5 in the second week, etc.) and the number of creatures available can be easily deduced.
> 
> And it's still not exact number. And rather than showing X + (Y -Z) I still prefer classical quantity description using human words which is the same throughout all series. "Many", "Not many", "Too many" etc, rather than maths exercices. smile I think. (I'm not offending, it's my personal point) that game's style will suffer from too excessive "numberisticness". Even game developers in the original prefered not to show spell damage (which is wrong in my point) to keep a balance within text messages and numerical values of stats, resource values and damage. If we full the game with all precise quantities using numbers it will loose some magic, I'm afraid.
> 
> I don't know, why am I so frightened with numbers... Could it be an injury from school lessons? But I loved maths and and always got "A". rofl
> 
> I completely understand and accept your proposal. It is more useful and rises comfort of the gameplay, comparing to what I am talking above. But I'm agree to sacrifice a little bit of preciseness of your proposal for the game style and "literary" pleasant-looking words. :)

I think that each of us will stay in our own position with different advantages and disadvantages in our eyes. :)

Why not propose this as an option or even an "on/off" setting in the fheroes2.cfg file?
So that all types of players are happy. :)

"Game: Display numbers instead of expressions for the quantity of creatures (https://github.com/ihhub/fheroes2/issues/1530), and for the exact range of creatures in the creature generators on the adventure map (https://github.com/ihhub/fheroes2/discussions/4665)."

> Just to annoy you guys even more, here's a picture of the official Japanese localization by Imagineer Co.:
> 
> ![image](https://user-images.githubusercontent.com/12501091/143295362-d0f81863-8306-4585-9ce5-d87aa0a24ecf.png)
> 
> Gotta love those 0-4 Ogre Lords. Maybe the hero has them, or maybe he doesn't - who knows? ... Technically, any army has 0-4 Black Dragons too.

For the Japanese localization, they understood the need for numbers for more "Strategy" oriented players compared to the terms for more "RPG" oriented players.

The "0" bug is funny.
We can also consider that the Ogre Lords are very dangerous even when there are none. :D

**Reply by Branikolog** [reply]
>Why not propose this as an option or even an "on/off" setting in the fheroes2.cfg file?
>So that all types of players are happy. :)

Not sure, there're lots of happy HoMMIII players now... 😨 
![unnamed](https://user-images.githubusercontent.com/55348946/143565464-fa8221bb-1d2f-42a8-bf63-d0e65a760d9e.png)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> > Why not propose this as an option or even an "on/off" setting in the fheroes2.cfg file?
> > So that all types of players are happy. :)
> 
> Not sure, there're lots of happy HoMMIII players now... :fearful:
> ![unnamed](https://user-images.githubusercontent.com/55348946/143565464-fa8221bb-1d2f-42a8-bf63-d0e65a760d9e.png)

I know that you don't like options and that you would say that there are too many options. :)
That's why I added the possibility of setting in the fheroes2.cfg file. :D

But in the case of numbers instead of terms, I consider this as a quality of life improvement (here there is a real demand), and not as an option to say, look we added one more option.

As for the number of options, I consider that when the player launches the game the first time, fheroes2 should be as close as possible to Heroes 2 (the improvements and corrections made to the original game are welcome and desirable (here the discussed option of numbers instead of terms is an improvement for me and not for you)), the available options being configured in this sense and that even if there are many.
Then, the player can make the **voluntary and desired step** to look at the available options and to modify them to "personalize" the game in a way that suits him better.
